src/app/pages/services/components/service-ssh/service-ssh.component.ts

```diff
@@ -1,180 +1,98 @@
-          tooltip: helptext.ssh_options_tooltip,
-        },
-      ],
-    },
-    { name: 'divider', divider: true },
-  ];
+  form = this.fb.group({
+    tcpport: [null as number],
+    rootlogin: [false],
+    passwordauth: [false],
+    kerberosauth: [false],
+    tcpfwd: [false],
+    bindiface: [[] as string[]],
+    compression: [false],
+    sftp_log_level: [null as SshSftpLogLevel],
+    sftp_log_facility: [null as SshSftpLogFacility],
+    weak_ciphers: [[] as SshWeakCipher[]],
+    options: [''],
+  });
 
-  advanced_field: string[] = [
-    'bindiface',
-    'compression',
-    'sftp_log_level',
-    'sftp_log_facility',
-    'options',
-    'weak_ciphers',
-  ];
+  readonly tooltips = {
+    tcpport: helptext.ssh_tcpport_tooltip,
+    rootlogin: helptext.ssh_rootlogin_tooltip,
+    passwordauth: helptext.ssh_passwordauth_tooltip,
+    kerberosauth: helptext.ssh_kerberosauth_tooltip,
+    tcpfwd: helptext.ssh_tcpfwd_tooltip,
+    bindiface: helptext.ssh_bindiface_tooltip,
+    compression: helptext.ssh_compression_tooltip,
+    sftp_log_level: helptext.ssh_sftp_log_level_tooltip,
+    sftp_log_facility: helptext.ssh_sftp_log_facility_tooltip,
+    weak_ciphers: helptext.ssh_weak_ciphers_tooltip,
+    options: helptext.ssh_options_tooltip,
+  };
 
-  custActions = [
-    {
-      id: 'basic_mode',
-      name: globalHelptext.basic_options,
-      function: () => {
-        this.fieldSets.find((set) => set.name === globalHelptext.fieldset_other_options).label = false;
-        this.fieldSets.find((set) => set.name === 'divider').divider = false;
-        this.isBasicMode = !this.isBasicMode;
-      },
-    },
-    {
-      id: 'advanced_mode',
-      name: globalHelptext.advanced_options,
-      function: () => {
-        this.fieldSets.find((set) => set.name === globalHelptext.fieldset_other_options).label = true;
-        this.fieldSets.find((set) => set.name === 'divider').divider = true;
-        this.isBasicMode = !this.isBasicMode;
-      },
-    },
-  ];
+  readonly sftpLogLevels$ = of(helptext.ssh_sftp_log_level_options);
+  readonly sftpLogFacilities$ = of(helptext.ssh_sftp_log_facility_options);
+  readonly sshWeakCiphers$ = of(helptext.ssh_weak_ciphers_options);
+  readonly bindInterfaces$ = this.ws.call('ssh.bindiface_choices').pipe(choicesToOptions());
 
-  protected ssh_bindiface: FormSelectConfig;
+  constructor(
+    private ws: WebSocketService,
+    private errorHandler: FormErrorHandlerService,
+    private cdr: ChangeDetectorRef,
+    private fb: FormBuilder,
+    private router: Router,
+  ) {}
 
-  isCustActionVisible(actionId: string): boolean {
-    if (actionId == 'advanced_mode' && !this.isBasicMode) {
-      return false;
-    } if (actionId == 'basic_mode' && this.isBasicMode) {
-      return false;
-    }
-    return true;
+  ngOnInit(): void {
+    this.ws.call('ssh.config').pipe(untilDestroyed(this)).subscribe((config) => {
+      this.form.patchValue(config);
+    });
```

Comment:
Display loading indicator during gets config
Add error handling

Reply:
And that's why we have code reviews :)